TFG-Labs/custom-shipping-step-by-items
# [WEB-478] BPF-2883 - Add Address Flow: Search Journey Editable Fields

### What problem is this solving?
When creating an address from **google auto-complete** in checkout`
- The current implementation logic for deleting co-ordinates from an address once edited needs to be removed`
-
- the primary auto populated fields will be locked as to prevent editing, Please see `Search` in the following table to know which addresses are locked https://www.notion.so/bashdotcom/Address-Flow-Update-914806a806274b538dd69568ba4bef82?pvs=4#d70f9dc928914eb1bdcbf66f7f985d84

#### How it works

**Deliver/utils**
- `checkForAddressResults`, `initGoogleAutocomplete`, `populateAddressFromSearch` `mapGoogleAddress` all moved to `Deliver/AddressSearch` - better co location of behavior and view 

**Deliver/FormComponents**
- `AddressSectionHeading`, `SubmitButton`, `ContactCard` - Ui components that are used accorss forms have been made reusable 

**Deliver/EditAddressForm**
- Changed to make use of of reusable   `FormComponents`

**Deliver/DeliverContainer**
- Logic moved to `AddressSearchSection`
- extension point to mount the `add-address-autocomplete-section`, for rendering the form 

**Deliver/AddressSearchSection**
- Renders the Google Autocomplete section

**Deliver/AddressSearch**
- Pulled out `AddressNotFoundNotification` into its own component
- moved `initGoogleAutocomplete` here - if the address from google is complete we go to the `add-address-autocomplete` form 
- moved `checkForAddressResults` here
- moved `populateAddressFromSearch`

**Deliver/AddAddressAutoCompleteForm**
Added a seperate form for when a result is returned from google auto complete
- Render the form 
- handle on form submit
- handle form validation and UI updates
- handle persistents to the orderform, master data, ui
- hadnles changing back to the correct view 

**DeliverController**
-> on view change clear the AddAddressAutoComplete form, and render it with the data from google autocomplete

#### How to test it?


### How to test it?
https://bpf2883--thefoschini.myvtex.com/

1. Go to check out from select address view click add address
<img width="594" alt="image" src="https://github.com/user-attachments/assets/048ade9a-6bbe-4267-af0e-c739d691c857">

2. You are presented with the auto complete view
<img width="601" alt="image" src="https://github.com/user-attachments/assets/0e3fafdd-48b0-4ff6-81ae-46e92f0c400c">


2.a If you if you enter an address that google returns incomplete data for (like no street) you are pushed into the manual flow with data prepopulated - stellenbosch central will take you to the manual flow

<img width="619" alt="image" src="https://github.com/user-attachments/assets/9701f8bf-3087-4ecd-acc0-437f76b50101">

2.b filling in an address where google auto complete returns all relevant fields will pull you into the edit flow: e.g. address 8 rigg street, mosel, uitenhage
<img width="607" alt="image" src="https://github.com/user-attachments/assets/c39a8195-d6a3-4744-af08-89f661324fc9"> 

3. On hitting save with invalid data - form validations witll kick in:
<img width="547" alt="image" src="https://github.com/user-attachments/assets/3dc47049-6eda-4391-bb6b-2aa4c6253735">



4 On Successful submission: the address overview will have selected your updated address and its update name/number will display
<img width="556" alt="image" src="https://github.com/user-attachments/assets/9abbe7e9-f379-4e51-82ad-6bca972b11e3">


5. The orderforms selected address would have also applied
https://bpf2883--thefoschini.myvtex.com/api/checkout/pub/orderForm/5280d0421d9f49e89ce85c31821bc537

<img width="736" alt="image" src="https://github.com/user-attachments/assets/599fa235-be6b-4f20-b057-4e522ba4cbe6">

<img width="648" alt="image" src="https://github.com/user-attachments/assets/93921605-c86e-40d3-a4ed-c27ae5f1ba6b">

6. Masterdata updated 
<img width="561" alt="image" src="https://github.com/user-attachments/assets/91e1cf9d-5e60-4c98-957a-8ef125cc6b38">



### Related to / Depends on
**JIRA**: https://tfginfotec.atlassian.net/browse/BPF-2883

#### How does this PR make you feel? 
Good
